code/game/mecha/equipment/tools/tools.dm

```diff
@@ -147,10 +147,20 @@
 
 	else if(istype(target, /turf/unsimulated/floor/asteroid)) //Digging for sand
 		if(do_after_cooldown(target, 1/3) && C == chassis.loc && src == chassis.selected)
+			var/count = 0
+			var/obj/structure/ore_box/ore_box = locate(/obj/structure/ore_box) in chassis:cargo

```

Comment:
Please use a variable or something pointing to an ore box in inventory rather than... whatever this is, please